book/content/reproducibility/04/barriers.md

```diff
@@ -40,9 +40,18 @@ This chapter outlines some of those barriers, and a couple of suggestions to get
 
 *replace this text with the content of barrier 7*
 
-## Barrier 8
-
-*replace this text with the content of barrier 8*
+## Big data and complex computational infrastructure
+
+Big data is conceptualised in different ways by different researchers. 
+The traditional understanding of big data focuses on complexity, variety of data sources, volume and velocity of the data. 
+It is extremely difficult to control to ensure consistency of data between the experiments. 
+In some cases, when data is sampled from a larger collection of data consistency cannot be guaranteed.
+
+A large variety of tools and products with a constantly changing ecosystem of supported technologies is available, meaning reproducing results in the future is highly dependent on the available technology at the time. 
+Some tools require in depth technical skills which are not widely available to data scientists. 
```

Comment:
I've just moved this section around a little, I hope that's ok?

Do you know of a link to the point about setting a random seed in tensorflow limiting parallelisation options? That's a great point!
```suggestion
Very often the results of statistical tests will vary depending on the configuration of the infrastructure that was used in each of the experiments, making it very hard to independently reproduce a result.
Experiments are often dependent on random initialisation for iterative algorithms and not all software includes the ability to fix a pseudorandom number without limiting parallelisation capabilities (for example  e.g. in Tensorflow).
These tools can require in depth technical skills which are not widely available to data scientists.
```